File: output/page_file_generator/Equfix_Place_On_Alert_selenium.py
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

class Equfix_Place_On_Alert_selenium:
    LOCATORS = {
        "continue_button": (By.XPATH, "//button[@id='continue-button']"),
        "zip": (By.XPATH, "//input[@id='zip']"),
        "state_dropdown": (By.XPATH, "//button[@id='efx-dropdown-label-753393']"),
        "city": (By.XPATH, "//input[@id='city']"),
        "address_line_2": (By.XPATH, "//input[@id='addressLine2Id']"),
        "address": (By.XPATH, "//input[@id='address']"),
        "phone_number": (By.XPATH, "//input[@id='phoneNumber']"),
        "ssn": (By.XPATH, "//input[@id='ssn']"),
        "dob": (By.XPATH, "//input[@id='dob']"),
        "last_name": (By.XPATH, "//input[@id='lastName']"),
        "first_name": (By.XPATH, "//input[@id='firstNameId']")
    }

    def __init__(self, driver):
        self.driver = driver
        self.wait = WebDriverWait(driver, 20)

    def wait_for_element(self, key):
        if key in self.LOCATORS:
            return self.wait.until(EC.presence_of_element_located(self.LOCATORS[key]))
        else:
            raise RuntimeError(f"Locator '{key}' not found in LOCATORS.")

    def click_element(self, key):
        element = self.wait_for_element(key)
        element.click()

    def enter_text(self, key, text):
        element = self.wait_for_element(key)
        element.clear()
        element.send_keys(text)

    def switch_to_new_window(self):
        if len(self.driver.window_handles) > 1:
            self.driver.switch_to.window(self.driver.window_handles[-1])

    def perform_actions(self):
        self.switch_to_new_window()
        self.click_element("continue_button")
        self.enter_text("ssn", "***-**-7575")
        self.click_element("last_name")
        self.enter_text("last_name", "test")
        self.click_element("phone_number")
        self.enter_text("phone_number", "768-676-****")
        self.click_element("dob")
        self.enter_text("dob", "04/22/1990")
        self.click_element("address")
        self.enter_text("address", "test")
        self.click_element("city")
        self.enter_text("city", "test")
        self.click_element("address_line_2")
        self.enter_text("address_line_2", "test")
        self.click_element("state_dropdown")
        self.enter_text("zip", "67567")

File: output/page_file_generator/Equfix_Place_On_Alert_selenium.java
import org.openqa.selenium.*;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.How;
import org.openqa.selenium.support.PageFactory;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.openqa.selenium.support.ui.ExpectedConditions;
import java.time.Duration;
import java.util.List;
import java.util.ArrayList;

public class Equfix_Place_On_Alert {

    private WebDriver driver;
    private WebDriverWait wait;

    @FindBy(how = How.XPATH, using = "//button[@id='continue-button']")
    private WebElement continueButton;

    @FindBy(how = How.XPATH, using = "//input[@id='zip']")
    private WebElement zipCode;

    @FindBy(how = How.XPATH, using = "//button[@id='efx-dropdown-label-753393']")
    private WebElement stateDropdown;

    @FindBy(how = How.XPATH, using = "//input[@id='city']")
    private WebElement cityName;

    @FindBy(how = How.XPATH, using = "//input[@id='addressLine2Id']")
    private WebElement addressLine2;

    @FindBy(how = How.XPATH, using = "//input[@id='address']")
    private WebElement addressLine1;

    @FindBy(how = How.XPATH, using = "//input[@id='phoneNumber']")
    private WebElement phoneNumber;

    @FindBy(how = How.XPATH, using = "//input[@id='ssn']")
    private WebElement ssn;

    @FindBy(how = How.XPATH, using = "//input[@id='dob']")
    private WebElement dateOfBirthMasked;

    @FindBy(how = How.XPATH, using = "//input[@id='lastName']")
    private WebElement lastName;

    @FindBy(how = How.XPATH, using = "//input[@id='firstNameId']")
    private WebElement firstNameId;

    public Equfix_Place_On_Alert(WebDriver driver) {
        this.driver = driver;
        this.wait = new WebDriverWait(driver, Duration.ofSeconds(20));
        PageFactory.initElements(driver, this);
    }

    private void waitForElement(WebElement element) {
        wait.until(ExpectedConditions.visibilityOf(element));
    }

    private void click(WebElement element) {
        waitForElement(element);
        element.click();
    }

    private void enterText(WebElement element, String text) {
        waitForElement(element);
        element.clear();
        element.sendKeys(text);
    }

    public void switchToNewWindow() {
        List<String> handles = new ArrayList<>(driver.getWindowHandles());
        if (handles.size() > 1) {
            driver.switchTo().window(handles.get(handles.size() - 1));
        }
    }

    public void clickContinueButton() {
        click(continueButton);
    }

    public void enterZipCode(String zip) {
        enterText(zipCode, zip);
    }

    public void clickStateDropdown() {
        click(stateDropdown);
    }

    public void enterCityName(String city) {
        enterText(cityName, city);
    }

    public void enterAddressLine2(String address) {
        enterText(addressLine2, address);
    }

    public void enterAddressLine1(String address) {
        enterText(addressLine1, address);
    }

    public void enterPhoneNumber(String phone) {
        enterText(phoneNumber, phone);
    }

    public void enterSSN(String socialSecurityNumber) {
        enterText(ssn, socialSecurityNumber);
    }

    public void enterDateOfBirth(String dob) {
        enterText(dateOfBirthMasked, dob);
    }

    public void enterLastName(String lastNameVal) {
        enterText(lastName, lastNameVal);
    }
}